### DISCUSSION TITLE
Replace delay by a proper pipeline

### DISCUSSION THREAD

**Discussion by toaster-code** [andesse/hue-clip-api.node-red-flows#20] [Ideas]
I would like to thank you first for this repository. Very interesting and I am already thinking about some improvements ideas. If I manage to do something myself I shall gladly contribute :)

Here is one idea: Replace the loop + delay in the core of the messaging by a proper pipeline with a counter to avoid congestion errors.
I think this is mostly trial and error to find a good threshold (originally HueMagic used 10 as maximum messages, without counting possible request from other sources like Hue application on smartphones) and try to see if there is an improvement in the design.

**Comment by andesse** [comment]
Thanks!

10 is the maximum for light request in the API. 
Smartphone relying on the exactly same API with same restrictions. 
You can create a developer account at Hue and have a look. 

I don’t know which flow you used, mine or V2. 
For me is the limit not necessary, cause I never trigger lights, I just use scenes. These are limited to one per second. I have an apartment with 5 rooms and never reached the limit, had an error.

In general I haven’t heard about any issues yet, following my recommendation. 

You can contribute your ideas, for sure. Export your flow and add it here as a comment. I will have a look next week.
